crl/updater/updater.go

```diff
@@ -335,6 +346,21 @@ func (cu *crlUpdater) tickShard(ctx context.Context, atTime time.Time, issuerNam
 	cu.log.Infof(
 		"Generating CRL shard: id=[%s] numChunks=[%d]", crlID, len(chunks))
 
+	if features.Enabled(features.LeaseCRLShards) {
+		// Notify the database that we're working on this shard.
+		deadline, ok := ctx.Deadline()
```

Comment:
This took me a while to understand what was going on. We're retrieving the amount of time left on the context to populate the remaining lease time in the `Until` field. This is almost sneaky.